academic-services/knowledge-exploration-service/concepts-queries.md

```diff
@@ -0,0 +1,113 @@
+
+This article describes how MAKES generates semantic interpretations of a natural language query that enable entities to be retrieved. It is intended for developers using MAKES who require a deeper understanding of the architecture and techniques used.
+
+## Architecture overview and diagram
+
+Processing a natural language query is a multi-stage process that starts with lexical analysis of the query, allowing the query to be rewritten for optimal search performance. The rewritten query is then parsed and interpretations hypothesized using a Context Sensitive Grammar (CSG). Valid hypothesis are resolved into full semantic query expressions which are in turn used to generate the complete interpretation response.
+
+:::image type="content" source="media/concept-semantic-interpretation-flow.png" alt-text="MAKES Semantic Interpretation Generation Architecture":::
+
+## Stage 1: Lexical analysis
+
+:::image type="content" source="media/concept-semantic-interpretation-lexical.png" alt-text="MAKES Semantic Interpretation Generation Architecture - Lexical Analysis":::
+
+Lexical analysis of the user query is done to help ensure that valid interpretation hypothesis can be generated.
+
+This is broke in to a number of different components.
+
+### Normalization
+
+Normalization transforms the query based on specific rules:
+
+* Removing non-essential words (stopwords, such as "the", "and", etc.)
+* Lower casing word letters to ensure uniformity with indeded data
+* Encoding specific types of academic concepts (i.e. equations, formulas, chemical compounds, etc.) to ensure they match the indexed data
+
+### Query operators
+
+MAKES supports a small handful of query operators that tell the grammar parser to perform a specific way when generating hypothesis:
+
+* '+' before a term indicates the term is required in any valid hypothesis
+* 'attributeName:' before a term indicate indicates the term can only be interpreted using values of that type, for example 'authorName:john platt' indicates that the terms 'john' and 'platt' can only be hypothesized as an author name(s)
+
+## Stage 2: Generate valid interpretation hypotheses
+
+:::image type="content" source="media/concept-semantic-interpretation-hypothesis.png" alt-text="MAKES Semantic Interpretation Generation Architecture - Generate interpretation hypothesis":::
+
+Semantic interpretation generation is broken into two important stages. The first of these is generating interpretation hypotheses using a [Context Sensitive Grammar (CSG)](https://en.wikipedia.org/wiki/Context-sensitive_grammar) conforming to the [Speech Recognition Grammar Specification (SRGS)](https://www.w3.org/TR/speech-grammar/).
+
+Generally SRGS grammars are context-free, meaning grammar rules can be matched lacking any additional context of previously matched rules. MAKES changes this behavior by including a *grammar context* that enables rules to reference metadata from previously matched rules.
```

Comment:
Add an example?

Reply:
The GetPapers part below is meant to be my example